src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -279,6 +278,12 @@ private DocumentMapper merge(DocumentMapper mapper) {
                     typeListener.beforeCreate(mapper);
                 }
                 mappers = newMapBuilder(mappers).put(mapper.type(), mapper).map();
+                if (mapper.parentFieldMapper().active()) {
+                    ImmutableSet.Builder<String> parentTypes = ImmutableSet.builder();
```

Comment:
maybe use a different local name for the builder, so it can't be confused with the member?